tests/init/neg/function1.scala

```diff
@@ -4,7 +4,7 @@ class Foo {
   val fun2: Int => Int = n => 1 + n + list.size
   fun2(5)
 
-  List(5, 9).map(n => 2 + n + list.size)         // error
+  List(5, 9).map(n => 2 + n + list.size)
```

Comment:
Only final vals of constant types (such as `false`, `2`, etc.) have different behavior due to const-folding.

We don't have a warning now, because at line 5, we already check the effect `this.list!`, thus the access `this.list!` is skipped here. If we comment out line 5, we will have a warning at this line.